vmm/src/igvm/mod.rs

```diff
@@ -0,0 +1,31 @@
+pub mod igvm_loader;
+use igvm_defs::IGVM_VHS_SNP_ID_BLOCK;
+use igvm_parser::snp_defs::SevVmsa;
+use std::mem::MaybeUninit;
+
+#[derive(Debug, Clone)]
+pub struct IgvmLoadedInfo {
+    pub gpas: Vec<u64>,
+    pub vmsa_gpa: u64,
+    pub first_gpa: u64,
+    pub snp_id_block: IGVM_VHS_SNP_ID_BLOCK,
+    pub start_gpa: u64,
+    pub length: u64,
+    pub vmsa: SevVmsa,
+}
+
+impl Default for IgvmLoadedInfo {
+    fn default() -> Self {
+        let ret = MaybeUninit::<IgvmLoadedInfo>::zeroed();
+        // SAFETY: set is initialized above
+        unsafe { ret.assume_init() }
```

Comment:
I don't think this is safe, as we don't know if `snp_id_block` and `vmsa` are properly initialized with `0` bytes (also I am not sure if `Vec<u64>` is properly initialized with `0` bytes). You can simply use the constructors for compound data types, right?

Reply:
All the fields are overwritten before use after parsing igvm file. IGVM_VHS_SNP_ID_BLOCK and SevVmsa don't derive default, that's why I have this default implementations. Regarding the Vec<u64> see the below new function we are actually using Vec::new there.

Reply:
This `Default` implementation is UB, so it needs to be fixed. Also, this UB can be leaked outside of this module that is even worse. You can do something like:

```
    fn default() -> Self {
        IgvmLoadedInfo{
            gpas: Vec::new(),
            vmsa_gpa: 0,
            first_gpa: 0,
            snp_id_block: FromZeroes::new_zeroed(),
            start_gpa: 0,
            length: 0,
            vmsa: FromZeroes::new_zeroed(),
        }
    }
```

Reply:
How about this one?
impl Default for IgvmLoadedInfo {
    fn default() -> Self {
        let uninit = MaybeUninit::<IgvmLoadedInfo>::zeroed();
        // SAFETY: uninit is initialized above
        unsafe { uninit.assume_init() }
    }
}

impl IgvmLoadedInfo {
    pub fn new() -> Self {
        IgvmLoadedInfo {
            gpas: Vec::new(),
            ..Default::default()
        }
    }
}